File: Algorithm/C++/942. DI String Match.cc
/*
 * Created on Fri Nov 23 2018 21:20:6
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */


// greedy
class Solution {
 public:
    vector<int> diStringMatch(string S) {
        vector<int> result;
        int left = 0, right = S.length();
        for (char c : S) {
          if (c == 'I') {
            result.push_back(left);
            left++;
          }
          else {
            result.push_back(right);
            right--;
          }
        }
        result.push_back(left);
        return result;
    }
};

File: Algorithm/Python/942. DI String Match.py
# -*- coding: utf-8 -*-
# Created on Fri Nov 23 2018 21:6:11
# Author: WuLC
# EMail: liangchaowu5@gmail.com

# greedy
class Solution(object):
    def diStringMatch(self, S):
        """
        :type S: str
        :rtype: List[int]
        """
        left, right = 0, len(S)
        result = []
        for i in xrange(len(S)):
            if S[i] == 'I':
                result.append(left)
                left += 1
            else:
                result.append(right)
                right -= 1
        assert left == right, 'left not equal to right at the last element'
        result.append(left)
        return result